File: 프로그래머스/unrated/140108. 문자열 나누기/문자열 나누기.py
def solution(s):
    tmp = s[0]
    a = 1
    b = 0
    answer = 0
    for i in range(1, len(s)):
        if a == b:
            tmp = s[i]
            a = 1
            b = 0
            answer += 1
            continue
        if tmp == s[i]:
            a += 1
        else:
            b += 1
    
    answer += 1
        
    return answer
        
            
        

File: 프로그래머스/1/140108. 문자열 나누기/문자열 나누기.java
class Solution {
    public int solution(String s) {
        int answer = 0;
        int i = 0;

        while (i < s.length()) {
            char x = s.charAt(i);
            int countX = 1;
            int countNotX = 0;
            i++;

            while (i < s.length()) {
                if (s.charAt(i) == x) {
                    countX++;
                } else {
                    countNotX++;
                }
                i++;
                if (countX == countNotX) {
                    break;
                }
            }

            answer++;
        }

        return answer;
    }
}
